packages/rest/test/integration/rest.server.integration.ts

```diff
@@ -245,20 +250,98 @@ servers:
       '/swagger-ui',
     );
     await server.get(RestBindings.PORT);
-    const url = new RegExp(
+    const urlString = new RegExp(
       [
         'http://petstore.swagger.io',
         '\\?url=http://\\d+.\\d+.\\d+.\\d+:\\d+/openapi.json',
       ].join(''),
     );
-    expect(response.get('Location')).match(url);
+    expect(response.get('Location')).match(urlString);
     expect(response.get('Access-Control-Allow-Origin')).to.equal('*');
     expect(response.get('Access-Control-Allow-Credentials')).to.equal('true');
   });
 
+  it('supports HTTPS protocol with key and certificate files', async () => {
+    const keyPath = path.join(__dirname, 'key.pem');
+    const certPath = path.join(__dirname, 'cert.pem');
+    const server = await givenAServer({
+      rest: {
+        port: 0,
+        protocol: 'https',
+        key: fs.readFileSync(keyPath),
+        cert: fs.readFileSync(certPath),
+      },
+    });
+    server.handler(({response}, sequence) => {
```

Comment:
Please extract the handler function to a helper, e.g. `dummyHandler` and fix all tests using the same dummy handler to use this new helper instead.